app/controllers/hub/tax_return_selections_controller.rb

```diff
@@ -30,9 +30,10 @@ def create
     end
 
     def new
-      @tr_ids = new_params[:tr_ids]
-      @client_count = Client.accessible_by(current_ability).distinct.joins(:tax_returns).where(tax_returns: { id: @tr_ids }).count
-      @tax_return_count = TaxReturn.accessible_by(current_ability).where(id: @tr_ids).count
+      @clients = Client.accessible_by(current_ability).with_eager_loaded_associations
+      @tr_ids = (new_params.dig(:create_tax_return_selection, :action_type) == "filtered-clients") ? TaxReturn.where(client: filtered_and_sorted_clients).pluck(:id) : new_params[:tr_ids]
```

Comment:
I think `all-filtered-clients` makes sense. I'll change that.

Reply:
After playing around with different options, we can't using `filtered_and_sorted_clients` to first define `@clients` because that method [itself needs a reference to `@clients` 😭](https://github.com/codeforamerica/vita-min/blob/df7a250bf82172d6617cae60d66d4e0c99af11b0/app/controllers/concerns/client_sortable.rb#L13). 

Looking at the mini-profiler, the callers are pretty small because their counts. I'm going to stick with this implementation for now.